DOCS-8202-easy-enable (#134)
mulesoft/docs-monitoring: modules/ROOT/pages/configure-monitoring-cloudhub.adoc

If you are using Maven for deployment, you must modify the `<muleVersion>`
tag in the `pom.xml` file. For example, you can change `<muleVersion>3.9.3</muleVersion>` to `<muleVersion>3.9.4</muleVersion>`).

== Enable Monitoring for Apps Deployed to CloudHub by Default

You can configure Anypoint Monitoring to enable monitoring by default for any newly deployed apps to CloudHub. When you enable the *(checkbox name)*, any Mule app you deploy to CloudHub has monitoring enabled by default.

This feature is available only if you have a Titanium subscription to Anypoint Platform, and you must use one of the following Mule runtime versions:

* 3.88 or later
* 3.9.x
* 4.x
// Check proper articulation for these versions. This list appears in the built-in dashboards UI.

> 

Author: ```suggestion
If you deployed Mule apps to CloudHub before you enabled this feature, you must enable monitoring for those apps manually. This feature does not retroactively enable monitoring for previously deployed apps.

```

If you are using Maven for deployment, you must modify the `<muleVersion>`
tag in the `pom.xml` file. For example, you can change `<muleVersion>3.9.3</muleVersion>` to `<muleVersion>3.9.4</muleVersion>`).

== Enable Monitoring for Apps Deployed to CloudHub by Default

You can configure Anypoint Monitoring to enable monitoring by default for any newly deployed apps to CloudHub. When you enable the *(checkbox name)*, any Mule app you deploy to CloudHub has monitoring enabled by default.

> This feature is available only if you have a Titanium subscription to Anypoint Platform, and you must use one of the following Mule runtime versions:

Reviewer: Correction:
This feature is available for everyone. If the customer has Titanium, the checkbox will be checked by default. It means all newly deployed apps will have monitoring on by default.

Author: ```suggestion
To use this feature, you must use one of the following Mule runtime versions:
```

> * 3.88 or later

Author: Actually, should I list these again? It's at the top of the topic.

Reviewer 2: No need to list the versions again as they are already listed above.

* 3.9.x
* 4.x
// Check proper articulation for these versions. This list appears in the built-in dashboards UI.

> If you deployed Mule apps to CloudHub before you enabled this feature, you must enable monitoring for those apps manually. This feature does not retroactively enable monitoring for previously deployed apps.

Author: ```suggestion
If you have a Titanium subscription to Anypoint Platform, this feature is enabled by default, and monitoring is enabled for all newly deployed Mule apps.

If you deployed Mule apps to CloudHub before you enabled this feature, you must enable monitoring for those apps manually. This feature does not retroactively enable monitoring for previously deployed apps.
```

If you are using Maven for deployment, you must modify the `<muleVersion>`
tag in the `pom.xml` file. For example, you can change `<muleVersion>3.9.3</muleVersion>` to `<muleVersion>3.9.4</muleVersion>`).

> == Enable Monitoring for Apps Deployed to CloudHub by Default

Author: ```suggestion
== Enable Automatic Monitoring for Apps Deployed to CloudHub
```

> You can configure Anypoint Monitoring to enable monitoring by default for any newly deployed apps to CloudHub. When you enable the *Enable Anypoint Monitoring for newly deployed CloudHub applications* check box, any Mule app that you deploy to CloudHub has monitoring enabled by default. 

Author: Option 1 makes the most sense with what we're releasing and whom we're trying to reach.

On 2, related to "if you do not have titanium," I thought we were trying not to document negatives? I also want to highlight the feature before noting for whom it's available. 

Option 3 is confusing with the last two lines; previously deployed apps and the titanium benefit are completely unrelated. They will have to go in and enable previously deployed apps regardless of their subscription level. This box applies only to newly deployed apps *after* it's enabled. The difference is that Titanium customers don't have to go in and enable the box when we release it; it's done for them, and THEN apps that are deployed to CH have Monitoring enabled. It doesn't automatically enable Monitoring for apps they'd deployed prior to this feature's release.

If you have a Titanium subscription to Anypoint Platform, this feature is enabled by default.

> . Sign in to Anypoint Platform and navigate to *Anypoint Monitoring*.

Author: ```suggestion
To configure automatic monitoring for each Mule app as it is deployed to CloudHub:

. Sign in to Anypoint Platform and navigate to *Anypoint Monitoring*.
```

. In the navigation menu, click *Settings*.
+
image::monitoring-settings-screen.png[]
+
. Select *CloudHub*.

> . Click *(checkbox name)*.

Author: ```suggestion
. Click *Enable Anypoint Monitoring for newly deployed CloudHub applications*.
```

If you deployed Mule apps to CloudHub before you enabled this feature, you must enable monitoring for those apps manually. This feature does not retroactively enable monitoring for previously deployed apps.

== Enable Monitoring for Apps Previously Deployed to CloudHub

> You can enable monitoring for Mule apps running on non-AM versions of Mule in the Anypoint Monitoring *Settings* page, the application's *Settings* page, or by using the Anypoint CLI for continuous integration and delivery (CI/CD).

Reviewer: The "non-AM" versions are no longer applicable. Let's take it out.

Author: If anyone is still behind on the times and running -AM versions, we should leave this. Otherwise, I'm happy to remove it.

Author: -AM was a suffix added to certain Runtime versions. It was always abbreviated in this manner. But I will follow up regarding edge cases of users who may or may not use them. We no longer support -AM runtimes at all, to my knowledge.

You can use the *Settings* page in Anypoint Monitoring to enable or disable apps deployed to CloudHub and running on Mule.

. Sign in to Anypoint Platform and navigate to *Anypoint Monitoring*.
. In the navigation menu, click *Settings*.
+
image::monitoring-settings-screen.png[]
+
. Select *CloudHub*.
. In the *Environment* list, select the environment where the application is running. +
A list of running applications is shown for the selected environment. +

> You can search for a specific application using the *List of resources* search box.

Author: No; searching within that space is highlighted because there are users with oodles of apps who are unlikely to see their app in that list immediately when it opens. As such, it is noted that searching will suggest apps based on the characters entered. It's not like our site search where they enter an entire term, hit enter, and *then* the list is populated. They only need a piece of the app name for this to work.

This was a good question, though. I didn't write these original docs, so I like it when you bring up stuff like this since we're actively cleaning up Monitoring.

> . Next to the application, click *Enable* to enable monitoring for the application, or *Disable* to disable monitoring for the application. +

Author: ```suggestion
. Next to the application, click *Enable* to enable monitoring for the application.
```

== Performance Impact

> Enabling monitoring can impact CPU utilization and memory utilization, which varies depending upon specific application configuration. You should test any performance impact in your pre-production environment prior to enabling monitoring in your production environment to ensure optimal performance and sizing.

Reviewer 3: Hmmm. I kind of worry about having this very important requirement isolated under its own title, rather than part of the task, and I wonder if there is some way we make this part of the intro text or a final step to the task or text after the final step of the task. 
Should we even consider retitling to Enable and Test...?

Author: We put it pretty low since it's a rather small percentage of customers with a pretty small impact overall, IIRC.
